src/python/pants/base/deprecated.py

```diff
@@ -156,6 +157,7 @@ def warn_or_error(
                   in a warning message.
   :param ensure_stderr: Whether use warnings.warn, or use warnings.showwarning to print
                         directly to stderr.
+  :param print_warning: Whether a warnings.WarningMessage object should be created.
```

Comment:
```suggestion
  :param print_warning: Whether to print a warning for deprecations *before* their removal. If this flag is off, an exception will still be raised for options past their deprecation date.
```
This is pretty specific, but I didn't quite feel like the relevance of the `WarningMessage` was super clear. Feel free to write whatever you'd like here, I just thought that it could maybe be described more concisely. If you want to keep this version, I'll approve it!

Reply:
That's fair, the extra specificity and context is much more helpful. Thx!